sky/utils/command_runner.py

```diff
@@ -767,6 +767,10 @@ def run(
         ]
         if self.context:
             kubectl_args += ['--context', self.context]
+        # If context is none, it means we are using incluster auth. In this
+        # case, need to set KUBECONFIG to /dev/null to avoid using kubeconfig.
+        if self.context is None:
+            kubectl_args += ['--kubeconfig', '/dev/null']
```

Comment:
Seems we have many places setting the context to some specific value for in-cluster auth. Will it cause that we are getting `in-cluster` or other context names instead of None?

Reply:
With this PR, it should be consistent now: context will always be set to the current context, except when running in-cluster (in which case context will be None and will need to be handled when passed to kubectl).

Reply:
Adding a comment to document this